lib/board.rb

```diff
@@ -0,0 +1,74 @@
+class Board
+  attr_accessor :cells
+  
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+  
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+  
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+  
+  def position(input)
+    cells[input.to_i - 1]
+  end
+  
+  def full?
+    cells.each do |position|
+      if position == " " || position == "" || position == nil
+        return false
+      else
+        
+      end
+    end
+    
+    return true
+    
+  end
+  
+  def turn_count
+    count = 0
+    cells.each do |position|
+      if position == " " || position == "" || position == nil
+        
+      else
+        count += 1
+      end
+    end
+    
+    count
+  end
+  
+  def taken?(input)
+    if cells[input.to_i - 1] == " " || cells[input.to_i - 1] == "" || cells[input.to_i - 1] == nil
+      return false
+    else
+      return true
+    end
+  end
+  
+  def valid_move?(input)
+    if input.to_i >= 1 && input.to_i <= 9
+      if taken?(input) == false
```

Comment:
You can simplify checking the valid integer range, and make it more semantic, by using [`Comparable#between?`](http://ruby-doc.org/core-2.2.3/Comparable.html#method-i-between-3F).
`position.between?(1,9)`